src/mbgl/actor/scheduler.cpp

```diff
@@ -1,22 +1,50 @@
 #include <mbgl/actor/scheduler.hpp>
-#include <mbgl/util/thread_local.hpp>
 #include <mbgl/util/thread_pool.hpp>
 
+#undef THREAD_LOCAL_UNSUPPORTED
+#ifdef __APPLE__
+# include <TargetConditionals.h>
+# include <Availability.h>
+# if defined(__IPHONE_OS_VERSION_MIN_REQUIRED) && __IPHONE_OS_VERSION_MIN_REQUIRED < __IPHONE_9_0
+#   define THREAD_LOCAL_UNSUPPORTED = 1
+# endif
+# if defined(__i386__) && defined(TARGET_OS_IPHONE)
+#   define THREAD_LOCAL_UNSUPPORTED = 1
+# endif
+#endif
+
+#ifdef THREAD_LOCAL_UNSUPPORTED
+#include <mbgl/util/thread_local.hpp>
+#endif
+
 namespace mbgl {
 
+#ifdef THREAD_LOCAL_UNSUPPORTED
 util::ThreadLocal<Scheduler> g_currentScheduler;
-
 static auto& current() {
     static util::ThreadLocal<Scheduler> scheduler;
     return scheduler;
 };
+#else
+namespace {
+    thread_local Scheduler* g_currentScheduler;
+} // namespace
+#endif
 
 void Scheduler::SetCurrent(Scheduler* scheduler) {
+#ifdef THREAD_LOCAL_UNSUPPORTED
     current().set(scheduler);
+#else
+    g_currentScheduler = scheduler;
```

Comment:
We aren't using this naming scheme (`g_` prefix) anywhere else in our code. Let's not introduce it here unless there's a really compelling reason to.